docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+    parameters:
+      location: westus
+      resourceGroup: default
+      sslEnforcement: disabled
+      firewallRules:
+      - name: "AllowAll"
+    	  startIPAddress: "0.0.0.0"
+    	  endIPAddress: "255.255.255.255"
+```
+
+
+
+#### Service Instance Management
+
+
+```console
+$ svcat describe instance mydb
+  Name:    	mydb
+  Namespace:    default
+  Status:  	Ready - The instance was provisioned successfully @
+                2017-11-30 13:11:49 -0600 CST
+  Type:    	mysql
+  Class:   	azure-mysql
+  Plan:    	basic50
+
+Parameters:
+  location: westus
+  resourceGroup: default
+  sslEnforcement: disabled
+  firewallRules:
+  - name: "AllowAll"
+    startIPAddress: "0.0.0.0"
+    endIPAddress: "255.255.255.255"
+```
+
+### Helm Charts
+
+
+#### Broker
+
+In the interim between this proposal and any upstream changes to the OSB API, brokers should include defaults for their Service Classes in their Helm charts.
+
+Note: The defaults must be created by Helm before Service Catalog performs a list against the broker because Helm cannot modify resources that it did not create. To ensure proper ordering, the ServiceClass resources should include a "pre-install" Helm hook.
+
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ClusterServiceClass
+metadata:
+  # Must match the name set during relist
+  name: b9978372-8dac-40ac-ae65-758b4a5075a5
+  # Ensure this the resource is created by Helm before broker list
+  annotations:
+    "helm.sh/hooks": pre-install # this needs to be deleted when the broker is deleted because helm won't cleanup this itself due to the hook
+  # indicates that this is a broker-provided class
+  ownerReferences:
+  - apiVersion: servicecatalog.k8s.io/v1beta1
+  	controller: true \
+Spec: \
+  externalId: b9978372-8dac-40ac-ae65-758b4a5075a5
+  # Only need specify new values not yet included in the OSB API
+  serviceType: mysql
+  defaultProvisionParameters:
+  	location: eastus
+  	resourceGroup: default
+  	sslEnforcement: disabled
+  	firewallRules:
+  	- name: "AllowAll"
+      startIPAddress: "0.0.0.0"
+      endIPAddress: "255.255.255.255"
+```
+
+
+
+#### Author
+
+Chart authors can use service types, instead of specifying specific broker's class, plan and parameters. Optional variables can be defined so an advanced user can specify a class/plan/parameters.
+
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceInstance
+metadata:
+  name: {{ template "fullname" . }}-mysql-instance
+spec:
+  serviceType: mysql-database-and-server
+  {{- if .Values.className != "" }}
+  clusterServiceClassName: {{ .Values.className }} # e.g. mysql
+  clusterServicePlanName: {{ .Values.planName }} # e.g. ha-db
+  {{- end}}
+  {{- if .Values.parameters != "" }}
+  {{ include "toYaml" .Values.parameters }}
+  {{- end}}
+
+```
+
+
+We will need to coordinate with OSB API to define an agreed upon set of service types. Below is just a suggestion:
```

Comment:
Perhaps Generic Broker Actions could solve this issue before it was an official part of the OSB spec.

Reply:
Oh, cool! I don't know much about generic broker actions. How do you think that could work?